### DISCUSSION TITLE
How to properly type params in options callback for React Navigation v7?

### DISCUSSION THREAD

**Discussion by pdrobyshev** [react-navigation/react-navigation#12143] [Q&A]
Hi! I'm trying out v7 navigation and love it so far but can't get the right types for route params in options callback. Hope anybody can help me

I use native stack navigator and have a screen. I want to pass `title` when navigating to this screen and get it from route params. 

```typescript
export const RootStackNavigator = createNativeStackNavigator({
  screens: {
    ...otherScreens,
    webview: {
      screen: WebviewScreen,
      options: ({ navigation, route }) => ({
        title: route.params.title,
      }),
    },
  },
})
```

At this point the `navigation` or `route` are not typed and I get error when trying to access anything inside `route.params`.
I could use `StaticScreenProps<>` to type `WebviewScreen` route params and then use `setOptions` to set the screen title, but is there a way to properly type options callback params?

**Comment by edwardhorsey** [comment]
Did you find out a way to do this?

**Reply by pdrobyshev** [reply]
Unfortunately I did not. Still have todo inside my code :/
